### DISCUSSION TITLE
fail2ban

### DISCUSSION THREAD

**Discussion by primski** [docker-mailserver/docker-mailserver#2845] [Q&A]
hello,

i see a lot of log in attempts that pass through fail2ban

> mailserver    | Oct 20 10:34:20 mail postfix/postscreen[928278]: CONNECT from [104.152.52.159]:54828 to [192.168.128.11]:25
> mailserver    | Oct 20 10:34:21 mail postfix/submission/smtpd[928279]: warning: hostname internettl.org does not resolve to address 104.152.52.159
> mailserver    | Oct 20 10:34:21 mail postfix/submission/smtpd[928279]: connect from unknown[104.152.52.159]
> mailserver    | Oct 20 10:34:21 mail postfix/submission/smtpd[928279]: SSL_accept error from unknown[104.152.52.159]: -1
> mailserver    | Oct 20 10:34:21 mail postfix/submission/smtpd[928279]: warning: TLS library problem: error:14209102:SSL routines:tls_early_post_process_client_hello:unsupported protocol:../ssl/statem/statem_srvr.c:1685:
> mailserver    | Oct 20 10:34:21 mail postfix/submission/smtpd[928279]: lost connection after STARTTLS from unknown[104.152.52.159]
> mailserver    | Oct 20 10:34:21 mail postfix/submission/smtpd[928279]: disconnect from unknown[104.152.52.159] ehlo=1 starttls=0/1 commands=1/2
> mailserver    | Oct 20 10:34:26 mail postfix/postscreen[928278]: PASS NEW [104.152.52.159]:54828
> mailserver    | Oct 20 10:34:26 mail postfix/smtpd[928298]: warning: hostname internettl.org does not resolve to address 104.152.52.159
> mailserver    | Oct 20 10:34:26 mail postfix/smtpd[928298]: connect from unknown[104.152.52.159]
> mailserver    | Oct 20 10:34:26 mail postfix/smtpd[928298]: lost connection after AUTH from unknown[104.152.52.159]
> mailserver    | Oct 20 10:34:26 mail postfix/smtpd[928298]: disconnect from unknown[104.152.52.159] auth=0/1 commands=0/1
> mailserver    | Oct 20 10:34:54 mail dovecot: imap-login: Disconnected (no auth attempts in 0 secs): user=<>, rip=104.152.52.216, lip=192.168.128.11, TLS, session=<DvUVM3PrvIlomDTY>
> mailserver    | Oct 20 10:34:54 mail dovecot: imap-login: Disconnected (no auth attempts in 0 secs): user=<>, rip=104.152.52.231, lip=192.168.128.11, session=<jT8WM3PrIshomDTn>
> mailserver    | Oct 20 10:34:55 mail dovecot: imap-login: Authenticate NTLM failed: Unsupported authentication mechanism.: user=<>, method=NTLM, rip=104.152.52.231, lip=192.168.128.11, TLS, session=<KW4kM3PrKMhomDTn>
> mailserver    | Oct 20 10:34:55 mail dovecot: imap-login: Disconnected (tried to use unsupported auth mechanism): user=<>, method=NTLM, rip=104.152.52.231, lip=192.168.128.11, TLS, session=<KW4kM3PrKMhomDTn>
> mailserver    | Oct 20 10:37:47 mail postfix/anvil[928281]: statistics: max connection rate 1/60s for (submission:104.152.52.159) at Oct 20 10:34:21
> mailserver    | Oct 20 10:37:47 mail postfix/anvil[928281]: statistics: max connection count 1 for (submission:104.152.52.159) at Oct 20 10:34:21
> mailserver    | Oct 20 10:37:47 mail postfix/anvil[928281]: statistics: max cache size 2 at Oct 20 10:34:26

Any chance we auto block these as well? 

thanks

**Comment by primski** [comment]
i thing the scammer changed the IP during the first and second attempt ...  ( i did a `fail2ban ban >>IP<<` manually  - this is why they changed IP )

can we just block the /16 mask instead of /32 ?

**Reply by casperklein** [reply]
The [feature](https://github.com/docker-mailserver/docker-mailserver/pull/2818) is not released yet. It will be in available in the next version.

For now: You can build the image yourself from the master branch or use the `:edge` tag to use that feature.

**Comment by primski** [comment]
great, many thanks .. i will wait for the release, no problem

thanks!

**Comment by DinosaurDad** [comment]
I'm also seeing a lot of login attempts. They're smart enough to space them out so they don't flood, but they keep trying the same bogus account names.

```
Jan 13 16:10:57 email postfix/smtps/smtpd[1871]: connect from unknown[183.102.31.115]
Jan 13 16:10:59 email postfix/smtps/smtpd[1871]: Anonymous TLS connection established from unknown[183.102.31.115]: TLSv1.2 with cipher ECDHE-RSA-AES128-GCM-SHA256 (128/128 bits)
Jan 13 16:11:03 email dovecot: auth: passwd-file(foo@xxxxxx.com,127.0.0.1,<2XLVqyvyHI5/AAAB>): unknown user (SHA1 of given password: 5f92bd)
Jan 13 16:11:05 email dovecot: imap-login: Aborted login (auth failed, 1 attempts in 2 secs): user=<lizard@mannus.com>, method=PLAIN, rip=127.0.0.1, lip=127.0.0.1, secured, session=<2XLVqyvyHI5/AAAB>
Jan 13 16:11:05 email postfix/smtps/smtpd[1871]: warning: unknown[183.102.31.115]: SASL LOGIN authentication failed: authentication failure
Jan 13 16:11:06 email postfix/smtps/smtpd[1871]: lost connection after AUTH from unknown[183.102.31.115]
Jan 13 16:11:06 email postfix/smtps/smtpd[1871]: disconnect from unknown[183.102.31.115] ehlo=1 auth=0/1 commands=1/2
```

I tried modifying the dovecot fail2ban regex to properly match the unknown user log entry, but it is trying to ban 127.0.0.1 instead of the real IP address. Any suggestions on how I can ban IP addresses that try to auth specific email addresses? This way I know they're bogus, without risk of banning legitimate user fat-fingering their password or something.

**Reply by DinosaurDad** [reply]
@casperklein any suggestions? The ban is being attempted on 127.0.0.1, but fortunately just results in `Ignore 127.0.0.1 by ignoreself rule`. I'm trying to understand how to get to the real IP address through docker mailserver.

**Comment by casperklein** [comment]
Not really. You might use a strict(er) [`fail2ban-jail.cf`](https://docker-mailserver.github.io/docker-mailserver/edge/config/security/fail2ban/) file:

```bash
cat fail2ban-jail.cf
[DEFAULT]

# "bantime" is the number of seconds that a host is banned.
bantime = 1w

# A host is banned if it has generated "maxretry" during the last "findtime"
# seconds.
findtime = 24h

# "maxretry" is the number of failures before a host get banned.
maxretry = 2

```
